File: ImageView/ImagePressClass/ImagePressClass/ImageScaleViewCtr.m
//
//  ImageScaleViewCtr.m
//  ImagePressClass
//
//  Created by EOC on 2017/4/16.
//  Copyright © 2017年   All rights reserved.
//

#import "ImageScaleViewCtr.h"

@interface ImageScaleViewCtr ()<UIScrollViewDelegate>{
    
    float _preScal;
}

@end

@implementation ImageScaleViewCtr

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"缩放";
    
//    _scrollview.maximumZoomScale = 5;
//    _scrollview.minimumZoomScale = 0.5;
    
    [self eocScale];
}

- (void)eocScale{
    
    //UIPanGestureRecognizer;
    UIPinchGestureRecognizer *pinchGesture = [[UIPinchGestureRecognizer alloc] initWithTarget:self action:@selector(pinchHandel:)];
    [_scrollview addGestureRecognizer:pinchGesture];
    
}

- (void)pinchHandel:(UIPinchGestureRecognizer*)gesture{
    
    NSLog(@"%f", gesture.scale);
    if (gesture.state == UIGestureRecognizerStateBegan) {
        _preScal = 1;
    }
    float scaleF = gesture.scale - _preScal + 1;
    _preScal = gesture.scale;
    _imageView.transform = CGAffineTransformScale(_imageView.transform, scaleF, scaleF);
    [_scrollview setContentSize:_imageView.frame.size];
    
    
    
    
}


- (UIView*)viewForZoomingInScrollView:(UIScrollView *)scrollView{
    
    return _imageView;
    
}





@end

File: ImagePressClass/ImagePressClass/Scale/ImageScaleViewCtr.swift
//
//  ImageScaleViewCtr.swift
//  ImagePressClass
//
//  Created by Casey on 14/01/2019.
//  Copyright © 2019 newplatform. All rights reserved.
//

import UIKit

class ImageScaleViewCtr: UIViewController, UIScrollViewDelegate {

    @IBOutlet var _scrollerView:UIScrollView?
    @IBOutlet var _imageView:UIImageView?
    override func viewDidLoad() {
        super.viewDidLoad()
        
        //_scrollerView?.maximumZoomScale = 5
     //   _scrollerView?.minimumZoomScale = 0.5
      
        
        bulidScale();
    }


    
    func bulidScale() {
        
        
        let pinchGesture = UIPinchGestureRecognizer.init(target: self, action: #selector(pinchHandle(gesture:)))
        
        _scrollerView?.addGestureRecognizer(pinchGesture)
        
    }
    
    
    var _preScale:CGFloat?
    @objc func pinchHandle(gesture: UIPinchGestureRecognizer) {
        
        if gesture.state == .began {
            
            _preScale = 1
        }
        let scaleCurrent = gesture.scale - _preScale! + 1
        _preScale = gesture.scale
        
        _imageView?.transform = _imageView!.transform.scaledBy(x: scaleCurrent, y: scaleCurrent)
        
        _scrollerView?.contentSize = _imageView!.frame.size
        
    }
    
    
    
    //MARK: scrollView delegate
    func viewForZooming(in scrollView: UIScrollView) -> UIView? {
        
        return _imageView
    }

}
